core/src/main/java/org/jdbi/v3/ExtensionRegistry.java

```diff
@@ -31,34 +31,38 @@ static ExtensionRegistry copyOf(ExtensionRegistry registry) {
         return new ExtensionRegistry(registry.factories);
     }
 
-    private static class Entry {
-        static Entry copyOf(Entry entry) {
-            return new Entry(entry.factory, entry.config.createCopy());
+    private static class Entry<C extends ExtensionConfig<C>> {
+        static <C extends ExtensionConfig<C>> Entry<C> copyOf(Entry<C> entry) {
+            return new Entry<>(entry.factory, entry.config.createCopy());
         }
 
-        final ExtensionFactory factory;
-        final ExtensionConfig config;
+        final ExtensionFactory<C> factory;
+        final C config;
 
-        Entry(ExtensionFactory factory, ExtensionConfig config) {
+        Entry(ExtensionFactory<C> factory, C config) {
             this.factory = factory;
             this.config = config;
         }
+
+        <E> E attach(Class<E> extensionType, Supplier<Handle> handle) {
+            return factory.attach(extensionType, config, handle);
+        }
     }
 
-    private final List<Entry> factories;
+    private final List<Entry<? extends ExtensionConfig<?>>> factories;
```

Comment:
I have mixed feelings on this one.

I originally wrote this Entry class with a generic parameter, but ended up removing it because all it did was add a bunch of noise to keep the type checker happy.

Case in point: the `<C>` parameter only has value inside the entry--the only value it provides is ensuring that the factory and the config types are compatible. We don't really need the wildcard parameter on this list, but Java barks at us anyway for REASONS.

This is really just a rant about the limitations of Java's generics. *sigh*

I like the new `Entry.attach` method though.